### ISSUE TITLE
[BUG] Combined measurements don´t work for version >= 0.29.0

### ISSUE THREAD

**Issue by MigCaSan** [PennyLaneAI/pennylane#4588]
### Expected behavior

For version 0.28.0 you can do a combined measurement like: "return [qml.expval(qml.PauliZ(wires=j)) for j in ancillaWires]".
This no longer works in newer versions.

### Actual behavior

When I run the same code in differents versions, two error messages may appear.

For >0.30.0 the error message look like this: "TypeError: NoneType takes no arguments"
For 0.29.0 the error message look like this: "TypeError: isclose(): argument 'other' (position 2) must be Tensor, not float"

But for 0.28.0 everything works fine.

### Tracebacks

```shell
res = type(self.tape._qfunc_output)(res)
TypeError: NoneType takes no arguments
```

### System information

```shell
Name: PennyLane
Version: 0.30.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning
Platform info:           Linux-6.2.0-32-generic-x86_64-with-glibc2.35
Python version:          3.10.12
Numpy version:           1.22.2
Scipy version:           1.11.1
Installed devices:
- default.gaussian (PennyLane-0.30.0)
- default.mixed (PennyLane-0.30.0)
- default.qubit (PennyLane-0.30.0)
- default.qubit.autograd (PennyLane-0.30.0)
- default.qubit.jax (PennyLane-0.30.0)
- default.qubit.tf (PennyLane-0.30.0)
- default.qubit.torch (PennyLane-0.30.0)
- default.qutrit (PennyLane-0.30.0)
- null.qubit (PennyLane-0.30.0)
- lightning.qubit (PennyLane-Lightning-0.30.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by josh146** [comment]
Thanks for reporting this @MigCaSan! Do you have a small minimal example of the code that generates this error? This would be useful for us in debugging this issue.

**Comment by MigCaSan** [comment]
Hi! Thanks for the respond. Yes, but i didn´t want to post it on GitHub. I will explain here the problema in more detail. I am creating a class which have a circuit as a private variable. I also modified the in-built method “call” to call there the circuit. The problem is when I do a single measurement works fine with all versions but when I tried a combined measurements it only works if I do in versions 0.28.0 or less. For greater versions one of the two errors describe in the post ocurs.class QCMM:    def __init__(self, input_wires, output_wires, layer=3, dev='default.qubit'):        self.input_wires = input_wires        self.output_wires = output_wires        self.layers = layer        self.device = qml.device(dev, wires=input_wires + output_wires)        weights = Variable(0.1 * torch.rand((18, int(np.log2(input_wires)))), requires_grad=True)        weights_dense_shape = qml.StronglyEntanglingLayers.shape(n_layers=2, n_wires=2)        weigthDense = Variable(0.1 * torch.rand(weights_dense_shape), requires_grad=True)        last_layer_weights = Variable(torch.rand((output_wires, 3)), requires_grad=True)        self.params = [weights, weigthDense, last_layer_weights]        def convolutional_layer(weights, wires, skip_first_layer=True):            """Adds a convolutional layer to a circuit.            Args:                weights (np.array): 1D array with 15 weights of the parametrized gates.                wires (list[int]): Wires where the convolutional layer acts on.                skip_first_layer (bool): Skips the first two U3 gates of a layer.            """            n_wires = len(wires)            assert n_wires >= 3, "this circuit is too small!"            for p in [0, 1]:                for indx, w in enumerate(wires):                    if indx % 2 == p and indx < n_wires - 1:                        if indx % 2 == 0 and not skip_first_layer:                            qml.U3(*weights[:3], wires=[w])                            qml.U3(*weights[3:6], wires=[wires[indx + 1]])                        qml.IsingXX(weights[6], wires=[w, wires[indx + 1]])                        qml.IsingYY(weights[7], wires=[w, wires[indx + 1]])                        qml.IsingZZ(weights[8], wires=[w, wires[indx + 1]])                        qml.U3(*weights[9:12], wires=[w])                        qml.U3(*weights[12:], wires=[wires[indx + 1]])        def pooling_layer(weights, wires):            """Adds a pooling layer to a circuit.            Args:                weights (np.array): Array with the weights of the conditional U3 gate.                wires (list[int]): List of wires to apply the pooling layer on.            """            n_wires = len(wires)            assert len(wires) >= 2, "this circuit is too small!"            for indx, w in enumerate(wires):                if indx % 2 == 1 and indx < n_wires:                    m_outcome = qml.measure(w)                    qml.cond(m_outcome, qml.U3)(*weights, wires=wires[indx - 1])        def conv_and_pooling(kernel_weights, n_wires, skip_first_layer=True):            """Apply both the convolutional and pooling layer."""            convolutional_layer(kernel_weights[:15], n_wires, skip_first_layer=skip_first_layer)            pooling_layer(kernel_weights[15:], n_wires)        @qml.qnode(self.device, interface="torch", diff_method='adjoint')        def conv_net(weights, weightsDense, last_layer_weights, features):            """Define the QCNN circuit            Args:                weights (np.array): Parameters of the convolution and pool layers.                last_layer_weights (np.array): Parameters of the last dense layer.                features (np.array): Input data to be embedded using AmplitudEmbedding."""            layers = weights.shape[1]            wires = list(range(self.input_wires))            ancillaWires = [x + input_wires for x in list(range(self.output_wires))]  # [12, 13, 14, 15, 16]            # inputs the state input_state            qml.AmplitudeEmbedding(features=features, wires=wires, pad_with=0.5, normalize=True)            qml.Barrier(wires=wires, only_visual=True)            # adds convolutional and pooling layers            for j in range(layers):                conv_and_pooling(weights[:, j], wires, skip_first_layer=(not j == 0))                wires = wires[::2]                qml.Barrier(wires=wires, only_visual=True)            # assert last_layer_weights.size(dim=0) == 4 ** (len(wires) + 3) - 1, (            #     "The size of the last layer weights vector is incorrect!"            #     f" \n Expected {4 ** (len(wires)+3) - 1}, Given {last_layer_weights.size}"            # )            # dense_layer(last_layer_weights, wires + [self.input_wires, self.input_wires+1, self.input_wires+2])            qml.StronglyEntanglingLayers(weightsDense, wires=wires)            for wireIdx, wire in enumerate(wires):                for idx, ancilla in enumerate(ancillaWires):                    qml.CNOT(wires=[wire, ancilla])                    # qml.ctrl(qml.RY, wire, control_values=1)(weightsRY[wireIdx, idx], wires=ancilla)            for idx, ancilla in enumerate(ancillaWires):                qml.U3(*last_layer_weights[idx, :], wires=ancilla)            return [qml.expval(qml.PauliZ(wires=j)) for j in ancillaWires]        self.circuit = conv_net    def __call__(self, features):        weights, weigthDense, last_layer_weights = self.params        return torch.nn.functional.softmax(self.circuit(weights, weigthDense, last_layer_weights, features))        # return self.circuit(weights, weightsRY, last_layer_weights,features)    def draw(self):        featuresRand = np.random.rand(2 ** self.input_wires)        weights, weightsRY, last_layer_weights = self.params        fig, ax = qml.draw_mpl(self.circuit, style='pennylane')(weights, weightsRY, last_layer_weights, featuresRand)        plt.show()  Enviado desde Correo para Windows De: Josh IzaacEnviado: lunes, 11 de septiembre de 2023 14:55Para: PennyLaneAI/pennylaneCC: MigCaSan; MentionAsunto: Re: [PennyLaneAI/pennylane] [BUG] Combined measurements don´t work for version >= 0.29.0 (Issue #4588) Thanks for reporting this @MigCaSan! Do you have a small minimal example of the code that generates this error? This would be useful for us in debugging this issue.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by MigCaSan** [comment]
Sorry, there you have.
class QCMM:
    def __init__(self, input_wires, output_wires, layer=3, dev='default.qubit'):
        self.input_wires = input_wires
        self.output_wires = output_wires
        self.layers = layer
        self.device = qml.device(dev, wires=input_wires + output_wires)

        weights = Variable(0.1 * torch.rand((18, int(np.log2(input_wires)))), requires_grad=True)
        weights_dense_shape = qml.StronglyEntanglingLayers.shape(n_layers=2, n_wires=2)
        weigthDense = Variable(0.1 * torch.rand(weights_dense_shape), requires_grad=True)
        last_layer_weights = Variable(torch.rand((output_wires, 3)), requires_grad=True)

        self.params = [weights, weigthDense, last_layer_weights]

        def convolutional_layer(weights, wires, skip_first_layer=True):
            """Adds a convolutional layer to a circuit.
            Args:
                weights (np.array): 1D array with 15 weights of the parametrized gates.
                wires (list[int]): Wires where the convolutional layer acts on.
                skip_first_layer (bool): Skips the first two U3 gates of a layer.
            """
            n_wires = len(wires)
            assert n_wires >= 3, "this circuit is too small!"

            for p in [0, 1]:
                for indx, w in enumerate(wires):
                    if indx % 2 == p and indx < n_wires - 1:
                        if indx % 2 == 0 and not skip_first_layer:
                            qml.U3(*weights[:3], wires=[w])
                            qml.U3(*weights[3:6], wires=[wires[indx + 1]])
                        qml.IsingXX(weights[6], wires=[w, wires[indx + 1]])
                        qml.IsingYY(weights[7], wires=[w, wires[indx + 1]])
                        qml.IsingZZ(weights[8], wires=[w, wires[indx + 1]])
                        qml.U3(*weights[9:12], wires=[w])
                        qml.U3(*weights[12:], wires=[wires[indx + 1]])

        def pooling_layer(weights, wires):
            """Adds a pooling layer to a circuit.
            Args:
                weights (np.array): Array with the weights of the conditional U3 gate.
                wires (list[int]): List of wires to apply the pooling layer on.
            """
            n_wires = len(wires)
            assert len(wires) >= 2, "this circuit is too small!"

            for indx, w in enumerate(wires):
                if indx % 2 == 1 and indx < n_wires:
                    m_outcome = qml.measure(w)
                    qml.cond(m_outcome, qml.U3)(*weights, wires=wires[indx - 1])

        def conv_and_pooling(kernel_weights, n_wires, skip_first_layer=True):
            """Apply both the convolutional and pooling layer."""
            convolutional_layer(kernel_weights[:15], n_wires, skip_first_layer=skip_first_layer)
            pooling_layer(kernel_weights[15:], n_wires)

        @qml.qnode(self.device, interface="torch", diff_method='adjoint')
        def conv_net(weights, weightsDense, last_layer_weights, features):
            """Define the QCNN circuit
            Args:
                weights (np.array): Parameters of the convolution and pool layers.
                last_layer_weights (np.array): Parameters of the last dense layer.
                features (np.array): Input data to be embedded using AmplitudEmbedding."""

            layers = weights.shape[1]
            wires = list(range(self.input_wires))
            ancillaWires = [x + input_wires for x in list(range(self.output_wires))]  # [12, 13, 14, 15, 16]

            # inputs the state input_state
            qml.AmplitudeEmbedding(features=features, wires=wires, pad_with=0.5, normalize=True)
            qml.Barrier(wires=wires, only_visual=True)

            # adds convolutional and pooling layers
            for j in range(layers):
                conv_and_pooling(weights[:, j], wires, skip_first_layer=(not j == 0))
                wires = wires[::2]
                qml.Barrier(wires=wires, only_visual=True)

            # assert last_layer_weights.size(dim=0) == 4 ** (len(wires) + 3) - 1, (
            #     "The size of the last layer weights vector is incorrect!"
            #     f" \n Expected {4 ** (len(wires)+3) - 1}, Given {last_layer_weights.size}"
            # )
            # dense_layer(last_layer_weights, wires + [self.input_wires, self.input_wires+1, self.input_wires+2])

            qml.StronglyEntanglingLayers(weightsDense, wires=wires)

            for wireIdx, wire in enumerate(wires):
                for idx, ancilla in enumerate(ancillaWires):
                    qml.CNOT(wires=[wire, ancilla])
                    # qml.ctrl(qml.RY, wire, control_values=1)(weightsRY[wireIdx, idx], wires=ancilla)

            for idx, ancilla in enumerate(ancillaWires):
                qml.U3(*last_layer_weights[idx, :], wires=ancilla)

            return [qml.expval(qml.PauliZ(wires=j)) for j in ancillaWires]

        self.circuit = conv_net

    def __call__(self, features):
        weights, weigthDense, last_layer_weights = self.params
        return torch.nn.functional.softmax(self.circuit(weights, weigthDense, last_layer_weights, features))
        # return self.circuit(weights, weightsRY, last_layer_weights,features)

    def draw(self):
        featuresRand = np.random.rand(2 ** self.input_wires)
        weights, weightsRY, last_layer_weights = self.params
        fig, ax = qml.draw_mpl(self.circuit, style='pennylane')(weights, weightsRY, last_layer_weights, featuresRand)
        plt.show()

**Comment by MigCaSan** [comment]
Sorry, there you have.

class QCMM:
    def __init__(self, input_wires, output_wires, layer=3, dev='default.qubit'):
        self.input_wires = input_wires
        self.output_wires = output_wires
        self.layers = layer
        self.device = qml.device(dev, wires=input_wires + output_wires)

        weights = Variable(0.1 * torch.rand((18, int(np.log2(input_wires)))), requires_grad=True)
        weights_dense_shape = qml.StronglyEntanglingLayers.shape(n_layers=2, n_wires=2)
        weigthDense = Variable(0.1 * torch.rand(weights_dense_shape), requires_grad=True)
        last_layer_weights = Variable(torch.rand((output_wires, 3)), requires_grad=True)

        self.params = [weights, weigthDense, last_layer_weights]

        def convolutional_layer(weights, wires, skip_first_layer=True):
            """Adds a convolutional layer to a circuit.
            Args:
                weights (np.array): 1D array with 15 weights of the parametrized gates.
                wires (list[int]): Wires where the convolutional layer acts on.
                skip_first_layer (bool): Skips the first two U3 gates of a layer.
            """
            n_wires = len(wires)
            assert n_wires >= 3, "this circuit is too small!"

            for p in [0, 1]:
                for indx, w in enumerate(wires):
                    if indx % 2 == p and indx < n_wires - 1:
                        if indx % 2 == 0 and not skip_first_layer:
                            qml.U3(*weights[:3], wires=[w])
                            qml.U3(*weights[3:6], wires=[wires[indx + 1]])
                        qml.IsingXX(weights[6], wires=[w, wires[indx + 1]])
                        qml.IsingYY(weights[7], wires=[w, wires[indx + 1]])
                        qml.IsingZZ(weights[8], wires=[w, wires[indx + 1]])
                        qml.U3(*weights[9:12], wires=[w])
                        qml.U3(*weights[12:], wires=[wires[indx + 1]])

        def pooling_layer(weights, wires):
            """Adds a pooling layer to a circuit.
            Args:
                weights (np.array): Array with the weights of the conditional U3 gate.
                wires (list[int]): List of wires to apply the pooling layer on.
            """
            n_wires = len(wires)
            assert len(wires) >= 2, "this circuit is too small!"

            for indx, w in enumerate(wires):
                if indx % 2 == 1 and indx < n_wires:
                    m_outcome = qml.measure(w)
                    qml.cond(m_outcome, qml.U3)(*weights, wires=wires[indx - 1])

        def conv_and_pooling(kernel_weights, n_wires, skip_first_layer=True):
            """Apply both the convolutional and pooling layer."""
            convolutional_layer(kernel_weights[:15], n_wires, skip_first_layer=skip_first_layer)
            pooling_layer(kernel_weights[15:], n_wires)

        @qml.qnode(self.device, interface="torch", diff_method='adjoint')
        def conv_net(weights, weightsDense, last_layer_weights, features):
            """Define the QCNN circuit
            Args:
                weights (np.array): Parameters of the convolution and pool layers.
                last_layer_weights (np.array): Parameters of the last dense layer.
                features (np.array): Input data to be embedded using AmplitudEmbedding."""

            layers = weights.shape[1]
            wires = list(range(self.input_wires))
            ancillaWires = [x + input_wires for x in list(range(self.output_wires))]  # [12, 13, 14, 15, 16]

            # inputs the state input_state
            qml.AmplitudeEmbedding(features=features, wires=wires, pad_with=0.5, normalize=True)
            qml.Barrier(wires=wires, only_visual=True)

            # adds convolutional and pooling layers
            for j in range(layers):
                conv_and_pooling(weights[:, j], wires, skip_first_layer=(not j == 0))
                wires = wires[::2]
                qml.Barrier(wires=wires, only_visual=True)

            # assert last_layer_weights.size(dim=0) == 4 ** (len(wires) + 3) - 1, (
            #     "The size of the last layer weights vector is incorrect!"
            #     f" \n Expected {4 ** (len(wires)+3) - 1}, Given {last_layer_weights.size}"
            # )
            # dense_layer(last_layer_weights, wires + [self.input_wires, self.input_wires+1, self.input_wires+2])

            qml.StronglyEntanglingLayers(weightsDense, wires=wires)

            for wireIdx, wire in enumerate(wires):
                for idx, ancilla in enumerate(ancillaWires):
                    qml.CNOT(wires=[wire, ancilla])
                    # qml.ctrl(qml.RY, wire, control_values=1)(weightsRY[wireIdx, idx], wires=ancilla)

            for idx, ancilla in enumerate(ancillaWires):
                qml.U3(*last_layer_weights[idx, :], wires=ancilla)

            return [qml.expval(qml.PauliZ(wires=j)) for j in ancillaWires]

        self.circuit = conv_net

    def __call__(self, features):
        weights, weigthDense, last_layer_weights = self.params
        return torch.nn.functional.softmax(self.circuit(weights, weigthDense, last_layer_weights, features))
        # return self.circuit(weights, weightsRY, last_layer_weights,features)

    def draw(self):
        featuresRand = np.random.rand(2 ** self.input_wires)
        weights, weightsRY, last_layer_weights = self.params
        fig, ax = qml.draw_mpl(self.circuit, style='pennylane')(weights, weightsRY, last_layer_weights, featuresRand)
        plt.show()

**Comment by MigCaSan** [comment]
class QCMM:
    def __init__(self, input_wires, output_wires, layer=3, dev='default.qubit'):
        self.input_wires = input_wires
        self.output_wires = output_wires
        self.layers = layer
        self.device = qml.device(dev, wires=input_wires + output_wires)

        weights = Variable(0.1 * torch.rand((18, int(np.log2(input_wires)))), requires_grad=True)
        weights_dense_shape = qml.StronglyEntanglingLayers.shape(n_layers=2, n_wires=2)
        weigthDense = Variable(0.1 * torch.rand(weights_dense_shape), requires_grad=True)
        last_layer_weights = Variable(torch.rand((output_wires, 3)), requires_grad=True)

        self.params = [weights, weigthDense, last_layer_weights]

        def convolutional_layer(weights, wires, skip_first_layer=True):
            """Adds a convolutional layer to a circuit.
            Args:
                weights (np.array): 1D array with 15 weights of the parametrized gates.
                wires (list[int]): Wires where the convolutional layer acts on.
                skip_first_layer (bool): Skips the first two U3 gates of a layer.
            """
            n_wires = len(wires)
            assert n_wires >= 3, "this circuit is too small!"

            for p in [0, 1]:
                for indx, w in enumerate(wires):
                    if indx % 2 == p and indx < n_wires - 1:
                        if indx % 2 == 0 and not skip_first_layer:
                            qml.U3(*weights[:3], wires=[w])
                            qml.U3(*weights[3:6], wires=[wires[indx + 1]])
                        qml.IsingXX(weights[6], wires=[w, wires[indx + 1]])
                        qml.IsingYY(weights[7], wires=[w, wires[indx + 1]])
                        qml.IsingZZ(weights[8], wires=[w, wires[indx + 1]])
                        qml.U3(*weights[9:12], wires=[w])
                        qml.U3(*weights[12:], wires=[wires[indx + 1]])

        def pooling_layer(weights, wires):
            """Adds a pooling layer to a circuit.
            Args:
                weights (np.array): Array with the weights of the conditional U3 gate.
                wires (list[int]): List of wires to apply the pooling layer on.
            """
            n_wires = len(wires)
            assert len(wires) >= 2, "this circuit is too small!"

            for indx, w in enumerate(wires):
                if indx % 2 == 1 and indx < n_wires:
                    m_outcome = qml.measure(w)
                    qml.cond(m_outcome, qml.U3)(*weights, wires=wires[indx - 1])

        def conv_and_pooling(kernel_weights, n_wires, skip_first_layer=True):
            """Apply both the convolutional and pooling layer."""
            convolutional_layer(kernel_weights[:15], n_wires, skip_first_layer=skip_first_layer)
            pooling_layer(kernel_weights[15:], n_wires)

        @qml.qnode(self.device, interface="torch", diff_method='adjoint')
        def conv_net(weights, weightsDense, last_layer_weights, features):
            """Define the QCNN circuit
            Args:
                weights (np.array): Parameters of the convolution and pool layers.
                last_layer_weights (np.array): Parameters of the last dense layer.
                features (np.array): Input data to be embedded using AmplitudEmbedding."""

            layers = weights.shape[1]
            wires = list(range(self.input_wires))
            ancillaWires = [x + input_wires for x in list(range(self.output_wires))]  # [12, 13, 14, 15, 16]

            # inputs the state input_state
            qml.AmplitudeEmbedding(features=features, wires=wires, pad_with=0.5, normalize=True)
            qml.Barrier(wires=wires, only_visual=True)

            # adds convolutional and pooling layers
            for j in range(layers):
                conv_and_pooling(weights[:, j], wires, skip_first_layer=(not j == 0))
                wires = wires[::2]
                qml.Barrier(wires=wires, only_visual=True)

            # assert last_layer_weights.size(dim=0) == 4 ** (len(wires) + 3) - 1, (
            #     "The size of the last layer weights vector is incorrect!"
            #     f" \n Expected {4 ** (len(wires)+3) - 1}, Given {last_layer_weights.size}"
            # )
            # dense_layer(last_layer_weights, wires + [self.input_wires, self.input_wires+1, self.input_wires+2])

            qml.StronglyEntanglingLayers(weightsDense, wires=wires)

            for wireIdx, wire in enumerate(wires):
                for idx, ancilla in enumerate(ancillaWires):
                    qml.CNOT(wires=[wire, ancilla])
                    # qml.ctrl(qml.RY, wire, control_values=1)(weightsRY[wireIdx, idx], wires=ancilla)

            for idx, ancilla in enumerate(ancillaWires):
                qml.U3(*last_layer_weights[idx, :], wires=ancilla)

            return [qml.expval(qml.PauliZ(wires=j)) for j in ancillaWires]

        self.circuit = conv_net

    def __call__(self, features):
        weights, weigthDense, last_layer_weights = self.params
        return torch.nn.functional.softmax(self.circuit(weights, weigthDense, last_layer_weights, features))
        # return self.circuit(weights, weightsRY, last_layer_weights,features)

    def draw(self):
        featuresRand = np.random.rand(2 ** self.input_wires)
        weights, weightsRY, last_layer_weights = self.params
        fig, ax = qml.draw_mpl(self.circuit, style='pennylane')(weights, weightsRY, last_layer_weights, featuresRand)
        plt.show()

**Comment by timmysilv** [comment]
hi @MigCaSan, nice find, and thanks for reporting this! You've hit a funny internal glitch that only happens with a very specific combination of things (using `qml.measure` as well as a qnode that returns a list of measurements, eg `[qml.expval(qml.PauliZ(i)) for i in range(2)]`).

Fortunately, if you instead return a tuple (eg. `tuple(qml.expval(qml.PauliZ(wires=j)) for j in ancillaWires)`), you don't hit this bug! I've also run your code and I see that `torch.nn.functional.softmax` doesn't like when you give it a tuple of torch tensors (as this code would do). I'd recommend changing your `__call__` function to use `hstack` to get around that like this:
```python
def __call__(self, features):
    weights, weigthDense, last_layer_weights = self.params
    res = self.circuit(weights, weigthDense, last_layer_weights, features)
    return torch.nn.functional.softmax(torch.hstack(res))
```

Hope that helps!

**Comment by MigCaSan** [comment]
Great! Thanks for the quick response and for the advice.

**Comment by dwierichs** [comment]
@timmysilv does this issue still exist? I tried reproducing with a much more minimal code and the combination you mentioned seemed fine, but maybe the above still errors out (not sure what the actual call to test is here?)

**Comment by timmysilv** [comment]
I'm not sure either 😅 I tried re-running it (swapping in `torch.tensor` in place of `Variable`) and giving `input_wires = output_wires = 2`, but it was complaining that the circuit was too small. I remember we had some funky handling near the end of `qml.execute` for `list` types specifically, so if that's gone then this issue probably doesn't exist anymore

**Comment by dwierichs** [comment]
:D Alright, thanks for looking into it! @josh146 do you have an opinion?
@MigCaSan do you happen to have a quick way of checking whether this bug is resolved for your original use case?

**Comment by MigCaSan** [comment]
Hi, I cannot check if the issue is solved right now, but in order to
replicate you only have to return different measurements from different
cables in a list instead of a tuple.

**Comment by josh146** [comment]
Thanks @MigCaSan! If we can't reproduce it with @timmysilv's minimal example any more, I would be okay to close this issue for now (as it likely is solved with the recent version of PL)

**Comment by dwierichs** [comment]
> in order to
replicate you only have to return different measurements from different
cables in a list instead of a tuple.

if this is the characterization, then the bug is indeed resolved :)

**Comment by josh146** [comment]
Okay, closing this issue for now. @MigCaSan don't hesitate to open another issue if you encounter another bug!

**Comment by MigCaSan** [comment]
Great! Thanks for your help
